src/client.rs

```diff
@@ -101,63 +140,72 @@ impl Client {
             client,
             key: api_key,
             cmd_tx,
+            background_tasks: Arc::new(Mutex::new(Vec::new())),
         };
 
         tokio::task::spawn(dispatcher.run());
 
         Ok(client)
     }
 
+    /// Kills all background tasks spawned by the client.
+    ///
+    /// This is useful when you want to stop all background tasks and clean up resources.
+    /// Note that this will abort all existing streams on this client instance.
+    pub fn kill_all_background_tasks(&self) {
+        for task in self.background_tasks.lock().unwrap().drain(..) {
+            task.abort();
+        }
+    }
+
     /// Broadcasts a signed transaction to the Fiber Network. Returns hash and the timestamp
     /// of when the first node received the transaction.
-    pub async fn send_transaction(&self, tx: TransactionSigned) -> Result<(String, i64)> {
-        let (res, rx) = oneshot::channel();
+    pub async fn send_transaction(&self, tx: TxEnvelope) -> FiberResult<(B256, i64)> {
+        let (response, rx) = oneshot::channel();
 
-        let _ = self
-            .cmd_tx
-            .send(SendType::Transaction { tx, response: res });
+        let cmd = SendType::Transaction { tx, response };
+        let _ = self.cmd_tx.send(cmd);
 
         let res = rx.await?;
 
-        Ok((res.hash.to_owned(), res.timestamp))
+        Ok((B256::from_str(&res.hash)?, res.timestamp))
     }
 
     /// Broadcasts a raw, RLP-encoded transaction to the Fiber Network. Returns hash and the timestamp
     /// of when the first node received the transaction.
-    pub async fn send_raw_transaction(&self, raw_tx: Vec<u8>) -> Result<(String, i64)> {
-        let (res, rx) = oneshot::channel();
+    pub async fn send_raw_transaction(&self, raw_tx: Vec<u8>) -> FiberResult<(B256, i64)> {
```

Comment:
Nit but let's use `TxHash` instead of `B256`?

Reply:
switched